src/Effects.js

```diff
@@ -1445,6 +1445,40 @@ module.exports = class Effects {
         }
       }
     };
+
+    this.music_wave = {
+      wave: [],
+      init: function () {
+        let wave = {
+          inc: 360/15,
+          heightFactor: 1,
+          yHalf: 300,
+          colorUnit: 400,
+          colorHalf: 200,
+        };
+        this.wave.push(wave);
+      },
+
+      draw: function (context) {
+        const centroid = context.centroid;
+        let scale = p5.map(centroid, 5000, 8000, 0, 1);
+        let a = 0;
+        p5.background('black');
+        const wave = this.wave[0];
+        for (let i = 1; i <= wave.colorUnit; i+=5) {
+          p5.stroke(lerpColorFromPalette(i/wave.colorUnit));
+          let yInitial = wave.yHalf - Math.abs(wave.yHalf * scale * (wave.heightFactor/wave.colorHalf) * p5.cos(a));
+          let yFinal = wave.yHalf + Math.abs(wave.yHalf * scale * (wave.heightFactor/wave.colorHalf) * p5.cos(a));
+          p5.line(i, yInitial, i, yFinal);
+          if ( i > wave.colorHalf) {
```

Comment:
From my interpretation, `wave.colorHalf` represents half of `wave.colorUnit`. If we can derive `wave.colorHalf` with `wave.colorUnit/2`, then I think we can get rid of `wave.colorHalf`.

Reply:
I changed some of the variable names to hopefully increase clarity.